### ISSUE TITLE
Fix: Switch the API to utilise dedicated endpoints

### ISSUE THREAD

**Issue by NickoOwen** [NickoOwen/bananabot#20]
Currently updates are sent from the frontend to the backend via a `POST` request to the `/admin` endpoint. ALL FORMS utilise this endpoint which is an issue for a few reasons.
* It is bad practice
* It becomes hard to maintain as more features and forms are added
* It doesn't allow for fetching individual bits of data (e.g. fetching the latest list of `announcements`) which makes AJAX either extremely difficult or straight up impossible

This task is to change this logic and create appropriate endpoints that can be used to `GET` and `POST` individual bits of data (e.g. `announcements`, `banana_time`, `selected_days` etc.). Flask has been a good framework to start with, however the alternative FastAPI framework may offer more support for implementing this functionality and should potentially be explored as part of this task.

**Done:**
* Individual API endpoints have been implemented for each bit of data the frontend may want to fetch in order to update without needing to refresh the page

**Comment by NickoOwen** [comment]
After some research and experimentation I have found that FastAPI solves a lot of issues that Flask was giving me troubles with. Specifically the conversion of data for request responses was far simpler using FastAPI. 

Without much effort I have switched the entire backend over to using the FastAPI framework while maintaining all the existing functionality. So far I have:
* Updated the UI to the new design created by @BasilDimopoulos (thanks for that - it looks great 😄)
  * Implemented the JavaScript for the timer functionality (mostly - minor tweaks still needed as this is a WIP issue still)
* Added password hashing to the HTTP Authentication
* Added JQuery and implemented new endpoints for requesting and sending data asynchronously in addition to having began implementing the logic for AJAX requests utilising JQuery

Still a fair bit to complete with this issue, however a lot of the ground work has been completed and most of what is left to do is implementing the AJAX logic for the rest of the admin page.

**Comment by NickoOwen** [comment]
I have now finished implementing all the forms using AJAX requests. I still need to update the `timer.js` file to fetch banana time from the backend as currently this does not happen. A bunch of tidying up etc. left to do still also.

I am currently debating where to put the AJAX functions for the forms. Due to JQuery's `.on()` method selecting all current AND future elements, it can be used outside of `$(document).ready` in these cases. I am unsure on what would be considered best practice in this case. Some resources I have found useful:
* https://www.w3schools.com/jquery/event_on.asp
* https://stackoverflow.com/questions/13062246/when-should-i-use-jquerys-document-ready-function

Once I come to a decision I will post another comment here. Any input is appreciated
